test/geometry_tests.cpp

```diff
@@ -10,156 +10,241 @@
 // Code to test
 #include "gillwald/geometry.hpp"
 
-namespace geometry {
-
-void printPixels(std::vector<std::vector<int>> pixels){
-	std::cout << "\n";
-	for (const auto &pixel:pixels ) {
-		std::cout << "(" << pixel[0] << ", " << pixel[1] << ")\n";		
-	}		
-	std::cout << "\n";
-}
-
-bool allXOrYCoorSame(int x_or_y, std::vector<std::vector<int>> pixels){
-	if (x_or_y == 0) { // Check X coordinates
-		x_or_y = 0;
-	} else { // Check Y coordinatesd
-		x_or_y = 1;
-	}
-
-	auto shouldBeEqualTo = pixels[0][x_or_y];
-
-	float sum = 0;
+using ::testing::ElementsAre;
+using ::testing::ElementsAreArray;
 
-	for (const auto &pixel:pixels) {
-		sum += pixel[x_or_y];
-	}
-
-	float avg = sum/static_cast<float>(pixels.size());
+namespace geometry {
 
-	if (avg == shouldBeEqualTo) {
-		return true;
-	} else {
-		return false;
+std::ostream& operator<<(std::ostream& os, const std::vector<Cell>& v){
+	os << "\n";
+	for (const auto& pixel:v) {
+		os << "(" << pixel.x << "," << pixel.y << ")\n";
 	}
-
+	os << "\n";
+	return os;
 }
 
-bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe, std::vector<std::vector<int>> v) {
-	std::vector<std::vector<int>>::iterator ptr;
-	if (firstOrLastPoint == 0) {
-		ptr = v.begin();
-	} else {
-		ptr = v.end() - 1;
-	}
-
-	if (((*ptr)[0] == x_shouldbe) && ((*ptr)[1] == y_shouldbe)) {
-		return true;
-	} else {
-		return false;
-	}
-
+inline bool operator==(const Cell &lhs,const Cell &rhs) {
+	return (lhs.x == rhs.x) && (lhs.y == rhs.y);
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_test) {
+	// Start and end points
+	const Cell start {0,0};
+	const Cell end {5,0};
+
 	// Get the points for the line connecting (0,0) and (5,0)
-	auto v = bresenham_conversion(0, 0, 5, 0);
+	const auto pixels = bresenham_conversion(start, end);
 
 	// Check if the vector size is 6 points 
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_EQ(pixels.size(), 6);
 
 	// Check if the first point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);
+	EXPECT_EQ(pixels.front(), Cell(0,0));
```

Comment:
Let's default to using braced initialization, unless required otherwise `Cell{0, 0}`

Reply:
Done

Reply:
actually, on review, when I make and run, I get this error:

/home/bilal/geometry/test/geometry_tests.cpp:53:37: error: macro "EXPECT_EQ" passed 3 arguments, but takes just 2
   53 |  EXPECT_EQ(pixels.front(), Cell{0,0});

So... should I just star with the parenthesis?

Reply:
let's stick with it for now

Reply:
https://stackoverflow.com/questions/50912219/gtest-false-negative-for-incorrect-number-of-macro-arguments